packages/hydrogen-edge-adapter/src/server.js

```diff
@@ -0,0 +1,123 @@
+// Virtual entry point for the app
+import * as remixBuild from '@remix-run/dev/server-build'
+import { createRequestHandler as netlifyCreateRequestHandler } from '@netlify/remix-edge-adapter'
+import { createRequestHandler as hydrogenCreateRequestHandler, getStorefrontHeaders } from '@shopify/remix-oxygen'
+import { createStorefrontClient, storefrontRedirect } from '@shopify/hydrogen'
+
+export async function getHydrogenClient({ env, request, waitUntil, caches, HydrogenSession, getLocaleFromRequest }) {
+  /**
+   * Open a cache instance in the worker and a custom session instance.
+   */
+  if (!env?.SESSION_SECRET) {
+    throw new Error('SESSION_SECRET environment variable is not set')
+  }
+
+  const [cache, session] = await Promise.all([
+    caches ? caches.open('hydrogen') : Promise.resolve(undefined),
+    HydrogenSession.init(request, [env.SESSION_SECRET]),
+  ])
+
+  /**
+   * Create Hydrogen's Storefront client.
+   */
+  const { storefront } = createStorefrontClient({
+    cache,
+    waitUntil,
+    i18n: getLocaleFromRequest(request),
+    publicStorefrontToken: env.PUBLIC_STOREFRONT_API_TOKEN,
+    privateStorefrontToken: env.PRIVATE_STOREFRONT_API_TOKEN,
+    storeDomain: env.PUBLIC_STORE_DOMAIN,
+    storefrontId: env.PUBLIC_STOREFRONT_ID,
+    storefrontHeaders: getStorefrontHeaders(request),
+  })
+
+  return {
+    storefront,
+    session,
+  }
+}
+
+export function createHydrogenHandler(overrides) {
+  return async function (request, env, executionContext) {
+    const waitUntil = (p) => executionContext.waitUntil(p)
+    const { storefront, session } = await getHydrogenClient(
+      Object.assign(
+        {
+          request,
+          env,
+          waitUntil,
+          caches,
+        },
+        overrides,
+      ),
+    )
+
+    const handler = await hydrogenCreateRequestHandler({
+      build: remixBuild,
+      mode: process.env.NODE_ENV,
+      getLoadContext: () => ({
+        session,
+        waitUntil,
+        storefront,
+        env,
+      }),
+    })
+
+    const response = await handler(request)
+
+    if (response.status === 404) {
+      /**
+       * Check for redirects only when there's a 404 from the app.
+       * If the redirect doesn't exist, then `storefrontRedirect`
+       * will pass through the 404 response.
+       */
+      return storefrontRedirect({ request, response, storefront })
+    }
+
+    return response
+  }
+}
+
+/**
+ * Export an Netlify Edge function handler in module format.
+ */
+export function createNetlifyEdgeHandler(overrides) {
+  return async function (request, context) {
+    // eslint-disable-next-line no-undef
+    const env = Netlify.env.toObject()
```

Comment:
I know you mentioned this existed the other day. I'm guessing this is preferred over `process.env` as it abstracts away the `process.env` and we can also enhance the `.env` with other stuff from our platform, or is there another reason to use it?

Reply:
I think it’d be fine to swap this instance to use `process.env`.

With the change to `packages/remix-edge-adapter/src/globals.ts` in this PR I’m trying to maintain existing compatibility with the existing `npm run dev` workflow (the Hydrogen dev server), which does not have the `Netlify` or `Deno` globals. This instance in particular only runs as part of the Netlify CLI though, so it would work with `process.env`, `Netlify.env`, or (least ideally imo) `Deno.env`.